client/packages/sprotty-client/src/features/tool-feedback/change-bounds-tool-feedback.ts

```diff
@@ -170,18 +299,64 @@ export class FeedbackMoveMouseListener extends MouseListener {
                     if (isMoveable(element) && isResizeable(element)) {
                         // If noElementOverlap Option is set perform collision detection
                         if (this.glspViewerOptions.noElementOverlap) {
+
```

Comment:
I find this very long block of code inside of a inline forEach very hard to read and mouseMove(...) is already a very long and technically deep method. I think it would be good to factor it out. After all, it handles a case which is optional (noElementOverlap configured), so it is a separate concern and should be in an own method/function. Also, I have the feeling this would be easy to do, since eventually we only add actions to the result, so the extracted methods could return actions if they want to, and here we only collect them.